File: frontend-app/profile-page/src/model/VehicleOptionsDTO.ts
import { LocationResponseDTO } from './LocationResponseDTO';

export class VehicleOptionsDTO {
    bodyTypes: string[];
    brands: string[];
    colors: string[];
    fuelTypes: string[];
    locations: LocationResponseDTO[];

    constructor(
        bodyTypes: string[],
        brands: string[],
        colours: string[],
        fuelTypes: string[],
        locations: LocationResponseDTO[]
    ) {
        this.bodyTypes = bodyTypes;
        this.brands = brands;
        this.colors = colours;
        this.fuelTypes = fuelTypes;
        this.locations = locations;
    }
}

File: api/vehicle-service/src/main/java/com/carrental/vehicleservice/model/dto/VehicleOptionsDTO.java
package com.carrental.vehicleservice.model.dto;

import lombok.Data;

import java.math.BigDecimal;
import java.util.Set;

@Data
public class VehicleOptionsDTO {

    private Set<String> bodyTypes;

    private Set<String> brands;

    private Set<String> colors;

    private Set<String> fuelTypes;

    private Set<LocationResponseDTO> locations;
}

